features/bootstrap/FeatureContext.php

```diff
@@ -6,39 +6,90 @@
 use WP_CLI\Process;
 use WP_CLI\Utils;
 
-// Inside a community package
-if ( file_exists( __DIR__ . '/utils.php' ) ) {
-	require_once __DIR__ . '/utils.php';
-	require_once __DIR__ . '/Process.php';
-	require_once __DIR__ . '/ProcessRun.php';
-	$project_composer = dirname( dirname( dirname( __FILE__ ) ) ) . '/composer.json';
-	if ( file_exists( $project_composer ) ) {
-		$composer = json_decode( file_get_contents( $project_composer ) );
-		if ( ! empty( $composer->autoload->files ) ) {
-			$contents = 'require:' . PHP_EOL;
-			foreach ( $composer->autoload->files as $file ) {
-				$contents .= '  - ' . dirname( dirname( dirname( __FILE__ ) ) ) . '/' . $file . PHP_EOL;
-			}
-			$folder = sys_get_temp_dir() . '/wp-cli-package-test';
-			if ( is_dir( $folder ) || mkdir( $folder ) || is_dir( $folder ) ) {
-				$project_config = "{$folder}/config.yml";
-				file_put_contents( $project_config, $contents );
-				putenv( 'WP_CLI_CONFIG_PATH=' . $project_config );
-			};
+/**
+ * Load required support files as needed before heading into the Behat context.
+ */
+function wpcli_bootstrap_behat_feature_context() {
+	// We try to detect the vendor folder in the most probable locations.
+	$vendor_locations = [
+		// wp-cli/wp-cli-tests is the root project.
+		dirname( dirname( __DIR__ ) ) . '/vendor',
+		// wp-cli/wp-cli-tests is a dependency.
+		dirname( dirname( dirname( __DIR__ ) ) ),
+	];
+
+	$vendor_folder = '';
+	foreach ( $vendor_locations as $location ) {
+		if ( is_dir( $location )
+		     && is_readable( $location )
+		     && is_file( "{$location}/autoload.php" ) ) {
```

Comment:
These two lines have space based precision alignment, please adjust this to ordinary tab indents.
```suggestion
			&& is_file( "{$location}/autoload.php" ) ) {
```

Reply:
Also for readability/ to distinguish more easily between the conditions and the code within the control structure, you may want to consider writing this like so:
```php
		if ( is_dir( $location )
			&& is_readable( $location )
			&& is_file( "{$location}/autoload.php" )
		) {
			$vendor_folder = $location;
			break;
		}
```

Reply:
They are not supposed to be indented. All three conditions are at the same indentation level, they are equals.

Reply:
> They are not supposed to be indented. All three conditions are at the same indentation level, 

Except they aren't, the second and third condition are indented by an additional five spaces (which should be one tab).

> they are equals.

That I get, but is not the point.

Reply:
the five spaces are alignment, like when you align assignments, only here we align boolean comparison operators. That's a common practice.

Reply:
Like here: https://github.com/php-fig-rectified/fig-rectified-standards/blob/master/PSR-2-R-coding-style-guide-additions.md#multi-line-declarationconditionconcatenation

Reply:
> Except they aren't, the second and third condition are indented by an additional five spaces (which should be one tab).

Spaces are not indentation. All three conditions are at the very same indentation level, with two of them having additional alignment (but not additional indentation). Changing the spaces into a tab turns them from alignment to indentation, so we 1 at level 3 and 2 at level 4.

Reply:
Ok, how about this:
```php
		if (
			is_dir( $location )
			&& is_readable( $location )
			&& is_file( "{$location}/autoload.php" )
		) {
			$vendor_folder = $location;
			break;
		}
```
That keeps them at the same indentation, but avoids the spaces.

Reply:
Hmm, I thought to remember something with regards to this discussion...
https://github.com/squizlabs/PHP_CodeSniffer/pull/1736

Reply:
I hear you, but the WP Coding Standards does not allow for alignment at the start of a line, only for indentation.

> Rule of thumb: Tabs should be used at the beginning of the line for indentation, while spaces can be used mid-line for alignment.

Ref: https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#indentation

> Ok, how about this:

That works :+1: (and as a side-bonus complies with PSR-12).